apsuite/commisslib/topup_feedbacks.py

```diff
@@ -0,0 +1,398 @@
+            if self._stopevt.is_set():
+                print('Stopping...')
+                break
+            print(egun.bias.voltage, egun.bias.voltage_mon)
+            egun.bias.set_voltage(bias)
+            _time.sleep(self.params.wait_each_point)
+
+        pvo.auto_monitor = False
+        pvo.remove_callback(cbv)
+
+        timing.evg.cmd_turn_off_injection(timeout=30)
+        egun.bias.set_voltage(ini_bias)
+        print('Measurement finished!')
+
+    def plot_and_fit_data(self):
+        """."""
+        pass
+
+
+class BiasFeedbackParams(_ParamsBaseClass):
+    """."""
+
+    def __init__(self):
+        """."""
+        super().__init__()
+        self.min_lifetime = 10 * 3600  # [s]
+        self.max_lifetime = 25 * 3600  # [s]
+        self.default_lifetime = 17 * 3600  # [s]
+        self.min_target_current = 99  # [mA]
+        self.max_target_current = 102  # [mA]
+        self.default_target_current = 100.0  # [mA]
+        self.coeffs_dcurr_vs_bias = [-50.0, 10.0]
+        self.min_delta_current = 0.000  # [mA]
+        self.max_delta_current = 1  # [mA]
+        self.min_bias_voltage = -52  # [V]
+        self.max_bias_voltage = -40  # [V]
+        self.ahead_set_time = 10  # [s]
+        self.use_gpmodel = False
+        self.gpmodel_lengthscale = 4  # [V]
+        self.gpmodel_variance = 0.16  # [mA^2]
+        self.gpmodel_noise_var = 0.03  # [mA^2]
+        self.gpmodel_max_num_points = 20
+        self.gpmodel_opt_each_pts = 10
+        self.gpmodel_bidimensional = False
+        self.gpmodel_issparse = False
+
+    def __str__(self):
+        """."""
+        dtmp = '{0:26s} = {1:9d}  {2:s}\n'.format
+        stmp = '{0:26s} = {1:9s}  {2:s}\n'.format
+        ftmp = '{0:26s} = {1:9.2f}  {2:s}\n'.format
+
+        stg = ''
+        stg += ftmp('min_lifetime', self.min_lifetime, '[s]')
+        stg += ftmp('max_lifetime', self.max_lifetime, '[s]')
+        stg += ftmp('default_lifetime', self.default_lifetime, '[s]')
+        stg += ftmp('min_target_current', self.min_target_current, '[mA]')
+        stg += ftmp('max_target_current', self.max_target_current, '[mA]')
+        stg += ftmp(
+            'default_target_current', self.default_target_current, '[mA]')
+        stg += ftmp('min_delta_current', self.min_delta_current, '[mA]')
+        stg += ftmp('max_delta_current', self.max_delta_current, '[mA]')
+        stg += ftmp('min_bias_voltage', self.min_bias_voltage, '[V]')
+        stg += ftmp('max_bias_voltage', self.max_bias_voltage, '[V]')
+        stg += ftmp('ahead_set_time', self.ahead_set_time, '[s]')
+        stg += stmp('coeffs_dcurr_vs_bias', str(self.coeffs_dcurr_vs_bias), '')
+        stg += stmp('use_gpmodel', str(self.use_gpmodel), '')
+        stg += dtmp(
+            'gpmodel_opt_each_pts', self.gpmodel_opt_each_pts,
+            ' (0 for no optimization)')
+        stg += ftmp('gpmodel_lengthscale', self.gpmodel_lengthscale, '[mA]')
```

Comment:
in __init__ the attributes have comments indicating its units are (lengthscale, variance, noise_var) = ([V], [mA^2], [mA^2) but in __str__ the corresponding units are ([mA], [V^2], [V^2]).